src/components/patch/PatchContainer.tsx

```diff
@@ -51,41 +57,61 @@ export class PatchContainer extends React.Component<Props, State> {
             />
           </Paper>
         </div>
-        <Patches />
+        <InfiniteScroll hasMore={this.state.hasMore} loadMore={this.loadPatches} initialLoad={false}>
+          <Patches />
+        </InfiniteScroll>
       </div>
     );
   }
 
-  private loadPatches() {
-    this.props.client.getPatches((err, resp, body) => {
-      const versions = ConvertToPatches(JSON.stringify(resp.body)).VersionsMap
-      this.setState({ 
-        versions: versions, 
-        visible: versions, 
-      });
-    }, this.props.client.username);
+  private loadPatches = () => {
+    if (this.state.hasMore) {
+      this.props.client.getPatches((err, resp, body) => {
+        const newPatches = Object.values(ConvertToPatches(resp.body).VersionsMap);
+        if (newPatches.length === 0) {
+          this.setState({ 
+            hasMore: false 
+          });
+          return;
+        }
+        // reorder the patches so that the newest element is at the front of the array
+        newPatches.reverse();
+        this.setState((prevState, props) => ({
+          pageNum: prevState.pageNum + 1,
+          allPatches: [... this.state.allPatches, ...newPatches],
+          visiblePatches: [... this.state.allPatches, ...newPatches],
+      })); 
+      }, this.props.client.username, this.state.pageNum);
+    }
+  }
+
+  private compareByDate(a: UIVersion, b: UIVersion) {
+    const dateA = moment(String(a.Version.create_time));
+    const dateB = moment(String(b.Version.create_time));
+    return dateB.isAfter(dateA) ? 1 : -1;
   }
 
   private search = (event: React.ChangeEvent<HTMLInputElement>) => {
     const query = event.currentTarget.value;
     const filteredPatches = this.filterItems(query);
     this.setState({
-      visible: filteredPatches,
+      visiblePatches: filteredPatches,
     });
   }
 
   private filterItems(query: string) {
-    const filtered = {};
+    const filtered: UIVersion[] = [];
     if (query === "") {
-      return this.state.versions;
+      return this.state.allPatches;
     }
-    Object.keys(this.state.versions).map(versionId => {
-      const patch = this.state.versions[versionId];
+    Object.keys(this.state.allPatches).map(versionId => {
+      const patch = this.state.allPatches[versionId];
       const description = patch.Version.message;
       if (description.toLowerCase().indexOf(query.toLowerCase()) !== -1) {
-        filtered[versionId] = patch;
+        filtered.push(patch);
       }
     });
+    filtered.sort(this.compareByDate);
```

Comment:
I don't think you need to do this. Javascript array methods like map iterate from the lowest index in the array, so what you have at this point should already be sorted

Reply:
Good to know- I did this originally because I wasn't totally sure that map would iterate from the lowest index. I have now removed that code.